client.go

```diff
@@ -148,6 +148,16 @@ func withParams(params map[string]string) requestOption {
 	}
 }
 
+func withPathParam(param string) requestOption {
+	return func(req *http.Request) error {
+		if len(param) == 0 {
+			return nil
+		}
+		req.URL.Path = fmt.Sprintf("%s/%s", req.URL.Path, param)
+		return nil
+	}
+}
```

Comment:
A `requestOption` with this name should likely accept a `map[string]string`, and perform replacement in the entire URL instead, so that it can be used with paths like

```
/api/v1/resource-a/{uuid-a}/resource-b
```

as well.

I don't think just appending of a single path segment justifies a dedicated `requestOption`.

Reply:
I don't understand what you suggest

Reply:
What I was trying to say is that the `withPathParam` function included in this PR doesn't make much sense to me, given its limitation (can only *append* one path segment).

The same result can be achieved with less code using `fmt.Sprintf`, in a more readable manner, e.g.

https://github.com/DependencyTrack/client-go/blob/a9bb24097d0bc91b8e6921f478305665a467ca1f/finding.go#L75

I was hinting that `withPathParam` *would* be useful, if it would support *named* placeholders in request paths, similar to [Chi's request parameters](https://github.com/go-chi/chi#url-parameters).

If that was the case, it would make URLs like this more readable:

https://github.com/DependencyTrack/client-go/blob/a9bb24097d0bc91b8e6921f478305665a467ca1f/finding.go#L109

by allowing for something like this:

```go
req, err := f.client.newRequest(ctx, http.MethodPost, "/api/v1/finding/project/{projectUUID}/analyze", withPathParams(map[string]string{
    "projectUUID": projectUUID,
}))
```

And that's just a really long way of saying I think you should just use `fmt.Sprintf` instead.

Reply:
thanks, now I understand what you mean.

Reply:
please take look